Applications/Spire/Include/Spire/Ui/TableModel.hpp

```diff
@@ -127,6 +149,53 @@ namespace Details {
   const T& TableModel::get(int row, int column) const {
     return std::any_cast<const T&>(at(row, column));
   }
+
+  template<typename F>
+  void TableModel::visit(const TableModel::Operation& operation, F&& f) const {
+    if(operation.type() == typeid(Transaction)) {
+      auto transaction = boost::get<Transaction>(&operation);
+      for(const auto& transaction_operation : transaction->m_operations) {
+        visit(transaction_operation, std::forward<F>(f));
+      }
+    } else {
+      if constexpr(std::is_invocable_v<std::decay_t<F>>) {
+        std::forward<F>(f)();
+      } else if constexpr(std::is_invocable_r_v<void, F, const Operation&>) {
+        std::forward<F>(f)(operation);
```

Comment:
Remove this clause as well.

Reply:
This part is used to deal with the default hanlder whose parameter is auto.